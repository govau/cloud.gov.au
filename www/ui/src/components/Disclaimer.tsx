import * as React from "react";
import styled, { css } from "../styled-components";
import { Flex, Box } from "grid-styled";

import * as star from "./star.svg";
import Container from "./Container";

const BrandLink = styled.a`
  text-decoration: none;
`;

const Star = styled.img`
  height: 1.25rem;
  margin-right: 0.5rem;
  vertical-align: middle;
`;

const Brand = styled.span`
  color: #ffffff;
  font-weight: 500;
  vertical-align: middle;
`;

const Official = styled.span`
  color: #ffffff;
  vertical-align: middle;
`;

const StyledDisclaimer = styled.div`
  ${({ theme }) =>
    css`
      background-color: ${theme.disclaimerColor};
    `};
`;

const RetiringTag = styled.span`
  ${({ theme }) =>
  css`
    background-color: #000;
    color: #ffffff;
    font-weight: bold;
    vertical-align: middle;
    padding:3px;
  `};
`;

const StyledRetiring = styled.div`
  ${({ theme }) =>
    css`
      background-color: #45c2f0;
    `};
`;

interface Props {}

const Disclaimer: React.StatelessComponent<Props> = () => (
  <>
    <StyledRetiring>
      <Container>
        <Flex wrap={true} pt={1} pb={1}>
          <Box w={[1, "auto"]} pr={2}>
            <BrandLink href="/">
              <RetiringTag>Important</RetiringTag>
            </BrandLink>
          </Box>
          <Box w={[1, "auto"]} mt={[1, 0]}>
            <Official>Cloud.gov.au will be decommissioned by September 2021. Contact our <a href="mailto:support@cloud.gov.au">support</a> team for any questions.</Official>
          </Box>
        </Flex>
      </Container>
    </StyledRetiring>
    <StyledDisclaimer>
      <Container>
        <Flex wrap={true} pt={1} pb={1}>
          <Box w={[1, "auto"]} pr={2}>
            <BrandLink href="/">
              <Star src={star} alt="" />
              <Brand>GOV.AU</Brand>
            </BrandLink>
          </Box>
          <Box w={[1, "auto"]} mt={[1, 0]}>
            <Official>Official Australian Government website</Official>
          </Box>
        </Flex>
      </Container>
    </StyledDisclaimer>
  </>
);

export default Disclaimer;
