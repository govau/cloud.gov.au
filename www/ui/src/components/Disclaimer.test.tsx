import * as React from "react";
import * as renderer from "react-test-renderer";

import Disclaimer from "./Disclaimer";

it("renders without crashing", () => {
  const tree = renderer.create(<Disclaimer />).toJSON();
  expect(tree).toMatchSnapshot();
});
