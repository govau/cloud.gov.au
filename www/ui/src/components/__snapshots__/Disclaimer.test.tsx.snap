// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c1 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
}

.c3 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-top: 8px;
}

.c4 {
  box-sizing: border-box;
  width: 100%;
  padding-right: 16px;
}

.c8 {
  box-sizing: border-box;
  width: 100%;
  margin-top: 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  max-width: 1024px;
}

.c5 {
  text-decoration: none;
}

.c6 {
  height: 1.25rem;
  margin-right: 0.5rem;
  vertical-align: middle;
}

.c7 {
  color: #ffffff;
  font-weight: 500;
  vertical-align: middle;
}

.c9 {
  color: #ffffff;
  vertical-align: middle;
}

@media screen and (min-width:40em) {
  .c4 {
    width: auto;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    width: auto;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    margin-top: 0px;
  }
}

<div
  className=""
>
  <div
    className="c0 c1"
    is={null}
  >
    <div
      className="c2 c3"
      is={null}
    >
      <div
        className="c4"
        is={null}
      >
        <a
          className="c5"
          href="/"
        >
          <img
            alt=""
            className="c6"
            src={
              Object {
                "0": "s",
                "1": "t",
                "2": "a",
                "3": "r",
                "4": ".",
                "5": "s",
                "6": "v",
                "7": "g",
                "default": "star.svg",
              }
            }
          />
          <span
            className="c7"
          >
            GOV.AU
          </span>
        </a>
      </div>
      <div
        className="c8"
        is={null}
      >
        <span
          className="c9"
        >
          Official Australian Government website
        </span>
      </div>
    </div>
  </div>
</div>
`;
